public/app/plugins/datasource/influxdb/types.ts

```diff
@@ -22,7 +29,14 @@ export interface InfluxQueryTag {
   value: string;
 }
 
+export enum InfluxQueryType {
+  Classic = 'Classic', // IFQL query builder
+  InfluxQL = 'InfluxQL', // raw ifql
+  Flux = 'Flux',
+}
+
 export interface InfluxQuery extends DataQuery {
+  queryType?: InfluxQueryType;
```

Comment:
Just my opinon... we should move the query builder to generating flux queries, as this will allow users to join on tags, something you cannot do in ifql. But more of a forward looking thing.